src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -76,15 +82,31 @@ func newBundleCommand(fs afero.Fs) *cobra.Command {
 			logsLimit, err := units.FromHumanSize(logsSizeLimit)
 			out.MaybeDie(err, "unable to parse --logs-size-limit: %v", err)
 
-			err = executeBundle(cmd.Context(), fs, cfg, cl, admin, logsSince, logsUntil, int(logsLimit), timeout)
+			// to execute the appropriate bundle we look for
+			// kubernetes_service_host env variable as an indicator that we are
+			// in a k8s environment
+			if _, ok := os.LookupEnv("KUBERNETES_SERVICE_HOST"); ok {
```

Comment:
This is what the official k8s client does to check if it's inside a k8s cluster:

https://github.com/kubernetes/client-go/blob/master/rest/config.go#L516

```
host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
	if len(host) == 0 || len(port) == 0 {
		return nil, ErrNotInCluster
	}
```

I'll add the check for the port once we rebase with the upstream PR